src/test/cli-integration/rbd/snap-diff.t

```diff
@@ -0,0 +1,47 @@
+  $ ceph osd pool create xrbddiff1 8
+  pool 'xrbddiff1' created
+  $ rbd create --thick-provision --size 1M xrbddiff1/xtestdiff1 --no-progress
+  $ rbd diff xrbddiff1/xtestdiff1 --format json
+  [{"offset":0,"length":1048576,"exists":"true"}]
+  $ rbd rm xrbddiff1/xtestdiff1 --no-progress
+  $ rbd create --size 1M xrbddiff1/xtestdiff1
+  $ rbd diff xrbddiff1/xtestdiff1 --format json
+  []
+  $ rbd snap create xrbddiff1/xtestdiff1 --snap=allzeroes
+  $ rbd diff xrbddiff1/xtestdiff1 --format json
+  []
+  $ rbd diff --from-snap=allzeroes xrbddiff1/xtestdiff1 --format json
+  []
+  $ rbd bench --io-type write --io-size 1M --io-total 1M xrbddiff1/xtestdiff1 > /dev/null 2>&1
+  $ rbd diff xrbddiff1/xtestdiff1 --format json
+  [{"offset":0,"length":1048576,"exists":"true"}]
+  $ rbd diff --from-snap=allzeroes xrbddiff1/xtestdiff1 --format json
+  [{"offset":0,"length":1048576,"exists":"true"}]
+  $ rbd snap create xrbddiff1/xtestdiff1 --snap=snap1
+  $ rbd snap list xrbddiff1/xtestdiff1 --format json | python -mjson.tool | sed 's/,$/, /'
```

Comment:
@smithfarm Now it fails here [1] because python2's json.tool sorts the keys while in python3 it is disabled by default [2].

[1] http://pulpito.ceph.com/trociny-2019-06-11_04:54:38-rbd-wip-mgolub-testing-distro-basic-smithi/
[2] https://docs.python.org/3/library/json.html#module-json.tool

Reply:
Good point. When migrating to python3 we'll have to add `--sort-keys`.